server/java/src/main/java/com/godaddy/asherah/grpc/AppEncryptionServer.java

```diff
@@ -0,0 +1,107 @@
+package com.godaddy.asherah.grpc;
+
+import com.godaddy.asherah.appencryption.SessionFactory;
+import com.google.common.util.concurrent.MoreExecutors;
+import io.grpc.ServerBuilder;
+import org.apache.commons.lang3.SystemUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+
+import io.grpc.Server;
+import io.grpc.netty.NettyServerBuilder;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import io.netty.channel.epoll.EpollServerDomainSocketChannel;
+import io.netty.channel.kqueue.KQueueEventLoopGroup;
+import io.netty.channel.kqueue.KQueueServerDomainSocketChannel;
+import io.netty.channel.unix.DomainSocketAddress;
+
+class AppEncryptionServer {
+  private static final Logger logger = LoggerFactory.getLogger(AppEncryptionServer.class);
+
+  private final Server server;
+  private final String udsFilePath;
+
+  /** Create an AppEncryptionServer server listening on default uds path using a {@code sessionFactory}. */
+  AppEncryptionServer(final SessionFactory sessionFactory) {
+    this(sessionFactory, Constants.DEFAULT_UDS_PATH);
+  }
+
+  /** Create an AppEncryptionServer server listening on uds path using a {@code sessionFactory}. */
+  AppEncryptionServer(final SessionFactory sessionFactory, final String udsFilePath) {
+    this.udsFilePath = udsFilePath;
+    NettyServerBuilder nettyServerBuilder = getNettyServerBuilder(udsFilePath);
+    Executor executor = MoreExecutors.directExecutor();
+    nettyServerBuilder.executor(executor);
+
+    server = nettyServerBuilder.addService(new AppEncryptionImpl(sessionFactory)).build();
+  }
+
+  /**
+   * Create an AppEncryptionServer server with {@code serverBuilder} listening on {@code udsFilePath}
+   * using a {@code sessionFactory}.
+   */
+  AppEncryptionServer(final SessionFactory sessionFactory, final String udsFilePath,
+      final ServerBuilder<?> serverBuilder) {
+    this.udsFilePath = udsFilePath;
+    server = serverBuilder.addService(new AppEncryptionImpl(sessionFactory)).build();
+  }
+
+  /** Start serving requests. */
```

Comment:
nit: add `@throws` tag

Reply:
decided to keep java comments like `// explanation comments`